lib/rspec/matchers/built_in/compound.rb

```diff
@@ -101,6 +123,11 @@ def matcher_supports_block_expectations?(matcher)
           false
         end
 
+        def matcher_list_for(matcher)
```

Comment:
> But BaseMatcher defines default diffable? as false. I assume user custom matcher does not have to derive from BaseMatcher?

We use `BaseMatcher` internally but have it labeled as private:

https://github.com/rspec/rspec-expectations/blob/45ff9faa26ea912d6b13d4044e5691b894d9fcc5/lib/rspec/matchers/built_in/base_matcher.rb#L9-L13